validator/src/main.rs

```diff
@@ -27,15 +27,13 @@ pub fn main() {
         ("init", _) => commands::run::execute(
             &matches,
             solana_version,
-            &ledger_path,
```

Comment:
To confirm - we don't want to call `create_and_canonicalize_directory()` at this level / for the other commands ? I think `run` can create a ledger whereas the other command should be operating on an existing one

Reply:
yeah, I think other commands should just 1) parse the ledger and 2) canonicalize the path, which will also verify its existence.

Reply:
> 2) canonicalize the path, which will also verify its existence.

We don't do that currently for other commands right ?

Reply:
it looks like right now they just`admin_rpc_service::connect(ledger_path)` without any sanity checks